styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -61,12 +61,19 @@ public void transitionInto(RunState state) {
     switch (state.state()) {
       case PREPARE:
         try {
-          final Event createdEvent =
-              Event.submit(state.workflowInstance(), getExecDescription(workflowInstance));
-          try {
-            stateManager.receive(createdEvent);
-          } catch (StateManager.IsClosed isClosed) {
-            LOG.warn("Could not send 'created' event", isClosed);
+          final Optional<Workflow> workflow = storage.workflow(workflowInstance.workflowId());
+          if (workflow.isPresent()) {
```

Comment:
I suspect this is still racy? It doesn't guarantee that ResourceNotFoundException won't occur in handling this state transition, does it? This is only partially a "peaceful way" - a code path is added without removing the objectionable (exception on non-existent workflow) path, so :-1:, for understandability it's better to have just one path even if dirty.

Fully peaceful handling of non-existent workflows would require effectively atomic state handling. (Or serializable, pardon my non-rigorous argument.) I believe we can have that, but that's an even more involved change.